src/librustc_resolve/diagnostics.rs

```diff
@@ -964,6 +964,31 @@ impl<'a> Resolver<'a> {
 
         err.emit();
     }
+
+    crate fn make_undeclared_type_suggestion(
+        &mut self,
+        ident: Ident,
+    ) -> (String, Option<Suggestion>) {
+        let parent_scope = &ParentScope::module(self.graph_root);
+        let typo_suggestion = self.early_lookup_typo_candidate(
+            ScopeSet::AbsolutePath(TypeNS),
```

Comment:
```suggestion
            ScopeSet::All(ns, false),
```
The `ns` is available from `resolve_path_with_ribs`.
The string produced by this function is currently dropped later for non-types due to implementation being a bit Frankenstein-y, but it's not a burden to produce the correct result.